update contribution manifesto for 4.x convergence onto api.ci prow (#831)
openshift/jenkins: README.md

    * https://access.redhat.com/containers/#/registry.access.redhat.com/openshift3/jenkins-slave-base-rhel7
    * https://access.redhat.com/containers/#/registry.access.redhat.com/openshift3/jenkins-agent-maven-35-rhel7
    * https://access.redhat.com/containers/#/registry.access.redhat.com/openshift3/jenkins-agent-nodejs-8-rhel7

To build a RHEL7 based image, you need to run Docker build on a properly
subscribed RHEL machine.

    ```
    $ git clone https://github.com/openshift/jenkins.git
    $ cd jenkins
    $ make build TARGET=rhel7 VERSION=2
    ```

Also note, as of 3.11, the RHEL images are hosted at registry.redhat.io as well.  This is the terms based 
registry and requires credentials for access.  See [Transitioning the Red Hat container registry](https://www.redhat.com/en/blog/transitioning-red-hat-container-registry) for details:
    * registry.redhat.io/openshift3/jenkins-2-rhel7:v3.11
    * registry.redhat.io/openshift3/jenkins-agent-nodejs-8-rhel7:v3.11
    * registry.redhat.io/openshift3/jenkins-agent-maven-35-rhel7:v3.11
    * registry.redhat.io/openshift3/jenkins-slave-base-rhel7:v3.11
    

> The openshift cluster install for 3.11 will insure that credentials are provided and subsequently available on the nodes

Reviewer: /s/insure/ensure/

Installation (OpenShift V4)
---------------------------------

> Starting with v4.0, the images are no longer hosted on DockerHub, but instead on Quay, for users without entitlements/subscriptions for support.  Their pull specs are:

Reviewer: ```suggestion
Starting with v4.0, the images are only available on quay.io for public community support.  Their pull specs are:
```

* quay.io/openshift/origin-jenkins-agent-maven:v4.0
* quay.io/openshift/origin-jenkins-agent-base:v4.0

The images are also still available at the Red Hat Container Catalog for customers with subscriptions, 
though with some changes in the naming.

As with the initial introduction in 3.11, given the [transitioning of the Red Hat container registry](https://www.redhat.com/en/blog/transitioning-red-hat-container-registry), the RHEL based images are available at both registry.access.redhat.com and registry.redhat.io.
The terms based registry, registry.redhat.io, which requires credentials for access, is the strategic direction, and 
will be the only location for RHEL8 based content when that is available.  The pull secret you obtain from try.openshift.com includes 
access to registry.redhat.io.  The image pull specs are:
* registry.redhat.io/openshift4/ose-jenkins:v4.0
* registry.redhat.io/openshift4/ose-jenkins-agent-nodejs:v4.0
* registry.redhat.io/openshift4/ose-jenkins-agent-maven:v4.0
* registry.redhat.io/openshift4/ose-jenkins-agent-base:v4.0

OpenShift v4 also removes the 32 bit JVM option.  Only 64 bit will be provided for all images.

> The `Dockerfile.rhel7` variants still exists, but as part of the `CentOS` vs. `RHEL` distinction no longer existing, the various `Dockerfile` files have been renamed to `Dockerfile.localdev` to more clearly denote that they are for builds on developers' local machines that most likely do not have a Red Hat subscription / entitlement.  The `Dockerfile.localdev` variants are structured to allow building of the images on machines without `RHEL` subscriptions, even though there base images are no longer based on `CentOS`.  And subscriptions are still required for use of `Dockerfile.rhel7`.

Reviewer 2: ```suggestion
The `Dockerfile.rhel7` variants still exists, but as part of the `CentOS` vs. `RHEL` distinction no longer existing, the various `Dockerfile` files have been renamed to `Dockerfile.localdev` to more clearly denote that they are for builds on developers' local machines that most likely do not have a Red Hat subscription / entitlement.  The `Dockerfile.localdev` variants are structured to allow building of the images on machines without `RHEL` subscriptions, even though the base images are no longer based on `CentOS`.  Subscriptions are still required building with `Dockerfile.rhel7`.
```

Author: part of the next push - thanks @grdryn
